sunxspex/sunxspex_fitting/data_loader.py

```diff
@@ -0,0 +1,860 @@
+                                                                                                                         new_counts=self.loaded_spec_data[spectrum]["extras"]["background_counts"],
+                                                                                                                         old_effective_exposures=self.loaded_spec_data[spectrum]["extras"]["original_background_effective_exposure"])
+            self.loaded_spec_data[spectrum]["extras"]["background_count_error"] = np.sqrt(self.loaded_spec_data[spectrum]["extras"]["background_counts"])
+            self.loaded_spec_data[spectrum]["extras"]["background_rate"] = self.loaded_spec_data[spectrum]["extras"]["background_counts"] / self.loaded_spec_data[spectrum]["extras"]["background_effective_exposure"] / self.loaded_spec_data[spectrum]["count_channel_binning"]
+            self.loaded_spec_data[spectrum]["extras"]["background_rate_error"] = np.sqrt(self.loaded_spec_data[spectrum]["extras"]["background_counts"]) / self.loaded_spec_data[spectrum]["extras"]["background_effective_exposure"] / self.loaded_spec_data[spectrum]["count_channel_binning"]
+
+    def group(self, channel_bins, counts, group_min):
+        """ Groups bins so they have at least a `group_min` number of counts.
+
+        Parameters
+        ----------
+        channel_bins, counts : np.array
+                Array of the channel bins and counts.
+
+        group_min : Int
+                The minimum number of counts allowed in a bin. This input is a starting number and then is checked
+                incrementally.
+                Default: None
+
+        Returns
+        -------
+        The number of counts left over at the end that could not be grouped into the minimum number and new bins/counts arrays.
+        """
+        binned_channel = []
+        binned_counts = []
+        combin = 0
+        reset_bin_counter = True
+        for c, count in enumerate(counts):
+            if count>=group_min and combin==0 and reset_bin_counter:
+                binned_channel.append(channel_bins[c])
+                binned_counts.append(count)
+            else:
+                if reset_bin_counter:
+                    start_e_bin = channel_bins[c][0]
+                    reset_bin_counter = False
+                combin += count
+                if combin >= group_min:
+                    binned_channel.append([start_e_bin, channel_bins[c][1]]) # starting at the last bin edge and the last edge of the bin we're on
+                    binned_counts.append(combin)
+                    combin = 0
+                    reset_bin_counter = True
+
+        return combin, binned_channel, binned_counts
+
+    def group_pha_finder(self, channels, counts, group_min=None, print_tries=False):
+        """ Takes the counts, and checks the bins left over from grouping the bins with a minimum value.
+
+        Parameters
+        ----------
+        channel, counts : np.array
```

Comment:
```suggestion
        channel, counts : `numpy.ndarray`
```
Imported types should be spelled out explicitly